src/pages/settings/Profile/PersonalDetails/DateOfBirthPage.tsx

```diff
@@ -79,58 +53,43 @@ function LegalNamePage({privatePersonalDetails}: LegalNamePageProps) {
     return (
         <ScreenWrapper
             includeSafeAreaPaddingBottom={false}
-            shouldEnableMaxHeight
-            testID={LegalNamePage.displayName}
+            testID={DateOfBirthPage.displayName}
         >
             <HeaderWithBackButton
-                title={translate('privatePersonalDetails.legalName')}
+                title={translate('common.dob')}
                 onBackButtonPress={() => Navigation.goBack()}
             />
             {isLoadingPersonalDetails ? (
                 <FullscreenLoadingIndicator style={[styles.flex1, styles.pRelative]} />
             ) : (
                 <FormProvider
                     style={[styles.flexGrow1, styles.ph5]}
-                    formID={ONYXKEYS.FORMS.LEGAL_NAME_FORM}
+                    formID={ONYXKEYS.FORMS.DATE_OF_BIRTH_FORM}
                     validate={validate}
-                    onSubmit={updateLegalName}
+                    onSubmit={PersonalDetails.updateDateOfBirth}
                     submitButtonText={translate('common.save')}
                     enabledWhenOffline
                 >
-                    <View style={[styles.mb4]}>
-                        <InputWrapper
-                            InputComponent={TextInput}
-                            inputID={INPUT_IDS.LEGAL_FIRST_NAME}
-                            name="lfname"
-                            label={translate('privatePersonalDetails.legalFirstName')}
-                            aria-label={translate('privatePersonalDetails.legalFirstName')}
-                            role={CONST.ROLE.PRESENTATION}
-                            defaultValue={legalFirstName}
-                            spellCheck={false}
-                        />
-                    </View>
-                    <View>
-                        <InputWrapper
-                            InputComponent={TextInput}
-                            inputID={INPUT_IDS.LEGAL_LAST_NAME}
-                            name="llname"
-                            label={translate('privatePersonalDetails.legalLastName')}
-                            aria-label={translate('privatePersonalDetails.legalLastName')}
-                            role={CONST.ROLE.PRESENTATION}
-                            defaultValue={legalLastName}
-                            spellCheck={false}
-                        />
-                    </View>
+                    <InputWrapper
+                        InputComponent={DatePicker}
+                        inputID={INPUT_IDS.DOB}
+                        label={translate('common.date')}
+                        defaultValue={privatePersonalDetails?.dob ?? ''}
+                        minDate={subYears(new Date(), CONST.DATE_BIRTH.MAX_AGE)}
+                        maxDate={subYears(new Date(), CONST.DATE_BIRTH.MIN_AGE)}
+                        onInputChange={() => {}}
+                        onTouched={() => {}}
```

Comment:
Are these always required props in InputWrapper? Not existed in js

Reply:
https://github.com/Expensify/App/pull/37426/files/daeb2a0ef88166fa05eb2f01d3490863eb82da7c#diff-b9bef912ad8e5897648f9d47e29cd0e412f86691d3c90d624756036722c75535

looks like they were from the deleted file @situchan

Reply:
@kubabutkiewicz please make `onInputChange`, `onTouched` optional in components/DatePicker/index.tsx. And remove these 2 lines

Reply:
Fixed 😄